pkg/core/native/notary.go

```diff
@@ -158,23 +159,31 @@ func (n *Notary) onPayment(ic *interop.Context, args []stackitem.Item) stackitem
 	if !additionalParams[0].Equals(stackitem.Null{}) {
 		to = toUint160(additionalParams[0])
 	}
-	till := toUint32(additionalParams[1])
-	currentHeight := ic.Chain.BlockHeight()
-	if till < currentHeight {
-		panic(fmt.Errorf("`till` shouldn't be less then the chain's height %d", currentHeight))
-	}
 
+	allowedChangeTill := ic.Tx.Sender() == to
+	currentHeight := ic.Chain.BlockHeight()
 	deposit := n.getDepositFor(ic.DAO, to)
+	till := toUint32(additionalParams[1])
 	if deposit == nil {
 		if amount.Cmp(big.NewInt(2*transaction.NotaryServiceFeePerKey)) < 0 {
 			panic(fmt.Errorf("first deposit can not be less then %d, got %d", 2*transaction.NotaryServiceFeePerKey, amount.Int64()))
 		}
 		deposit = &state.Deposit{
 			Amount: new(big.Int),
 		}
+		if !allowedChangeTill {
+			till = currentHeight + defaultDepositDeltaTill
+		}
 	} else {
-		if till < deposit.Till {
-			panic(fmt.Errorf("`till` shouldn't be less then the previous value %d", deposit.Till))
+		if allowedChangeTill { // only deposit's owner is allowed to set or update `till`
+			if till < currentHeight {
```

Comment:
It should either be a generic (relative to `deposit == nil`) check or not exist at all, it doesn't affect correctness of this contract and it's hard to tell what is better for the user here, failed or partially successful (deposit accepted, but not really usable) transaction.

Reply:
So then maybe we'll set the default `till` value in `onPayment` method, and then the owner will be able to update `till` value via `lockDepositUntil`?

Reply:
The majority of invocations will have `to == from` and in this case it's quite natural to allow setting `till`.

Reply:
OK, I rewrote this part, now `till` check is generic.